tracing-journald/src/lib.rs

```diff
@@ -257,12 +286,12 @@ fn put_metadata(buf: &mut Vec<u8>, meta: &Metadata, span: Option<usize>) {
     }
 }
 
-fn put_debug(buf: &mut Vec<u8>, name: &str, value: &dyn fmt::Debug) {
+fn put_field_value(buf: &mut Vec<u8>, name: &str, write_value: impl FnOnce(&mut Vec<u8>)) {
```

Comment:
it could _maybe_ be worth a comment on what this function does, now that the relationship between the various `put_XXX` functions is a bit more complex? but, i don't think it's necessary.

Reply:
@hawkw @Ralith I've documented this function, and I've also renamed this one and `put_field` to clarify the difference between these two.

I hope that's okay; if the refactoring goes too far I'm happy to revert these changes.